src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala

```diff
@@ -0,0 +1,97 @@
+package mesosphere.marathon.core.matcher.base.util
+
+import mesosphere.marathon.core.task.Task.LocalVolume
+import mesosphere.marathon.state.PathId
+import org.apache.mesos.Protos.Resource.ReservationInfo
+import org.apache.mesos.{ Protos => Mesos }
+import org.slf4j.LoggerFactory
+
+class OfferOperationFactory(
+    private val principalOpt: Option[String],
+    private val roleOpt: Option[String]) {
+
+  private[this] val log = LoggerFactory.getLogger(getClass)
+
+  // FIXME (217): log.warn in these cases? or only if we really need and access them?
+
+  private[this] lazy val principal: String = principalOpt match {
+    case Some(value) => value
+    case _ =>
+      log.error("No principal set. See --mesos_authentication_principal")
+      "n/a"
```

Comment:
As discussed, this will now throw if accessed and not defined – since we don't know whether a Marathon user will use this feature, we can only validate the flags when POSTing an AppDefinition and reserving/creating.